src/1-object-oriented-programming-and-object-oriented-design/Triangle.ts

```
@@ -0,0 +1,50 @@
+import { Shape } from './Shape';
+import { Point } from './Point';
+
+const enum TriangleTypes {
+  Equilateral = "equilateral triangle",
+  Isosceles = "isosceles triangle",
+  Scalene = "scalene triangle"
+}
+
+export class Triangle extends Shape {
+  constructor(points: Point[])
+  constructor(points: Point[], color: string, filled: boolean)
+
+  constructor(points: Point[], color?: string, filled?: boolean) {
+    super(points, color as any, filled as any);
+
+    this.points = points.slice(0, 3);
```

I would throw an error if `points.length > 3`